### PR TITLE
fix(excel2json): don't crash if json_header.xlsx is present when using the default command (RDU-24)

### PR DESCRIPTION
**Pull request by Nora-Olivia-Ammann**

### CODE REVIEW

File: src/dsp_tools/commands/excel2json/project.py [hunk 1]
```diff
@@ -59,7 +59,7 @@ def excel2json(
 def _validate_folder_structure_get_filenames(data_model_files: str) -> tuple[list[Path], list[Path]]:
     if not Path(data_model_files).is_dir():
         raise UserError(f"ERROR: {data_model_files} is not a directory.")
-    folder = [x for x in Path(data_model_files).glob("*") if _non_hidden(x)]
+    folder = [x for x in Path(data_model_files).glob("*") if _non_hidden(x) and x.is_dir()]
```
Comment (jnussbaum) [inline src/dsp_tools/commands/excel2json/project.py:63]: This list contains all folders, i.e. the list folder + the onto folders. So it should be called "list_folder_and_onto_folders", or it that's too long, "folders".
It should then also be changed on these lines:
- 80
- 139
- 150

The current name has now really hindered me to understand what is going on 😅

File: src/dsp_tools/commands/excel2json/project.py [hunk 2]
```diff
@@ -23,11 +23,12 @@ def excel2json(
...
         data_model_files
         |-- lists
-        |   |-- de.xlsx
-        |   `-- en.xlsx
+        |   |-- list.xlsx
+        |   `-- list_2.xlsx
         `-- onto_name (onto_label)
             |-- properties.xlsx
             `-- resources.xlsx
+        | -- json_header.xlsx (optional)
```
Comment (jnussbaum) [inline src/dsp_tools/commands/excel2json/project.py:31]: This is not a correct tree

File: src/dsp_tools/commands/excel2json/project.py [hunk 3]
```diff
@@ -104,6 +107,7 @@ def new_excel2json(
...
```
Comment (jnussbaum) [inline src/dsp_tools/commands/excel2json/project.py:110]: This is not correct

File: src/dsp_tools/commands/excel2json/project.py (hunk 2 above)
Comment (Nora-Olivia-Ammann) [reply src/dsp_tools/commands/excel2json/project.py:31]: then I don't know how it should look like.

File: docs/file-formats/excel2json.md [hunk 4]
```diff
@@ -28,12 +28,12 @@ The following folder structure has to be used, when invoking the standard `excel
...
-`-- onto_name (onto_label)
-    |-- properties.xlsx
-    `-- resources.xlsx
-    `json_header.xlsx` (optional)
+|    |-- de.xlsx
+|    |-- en.xlsx
+|-- onto_name (onto_label)
+|    |-- properties.xlsx
+|    |-- resources.xlsx
+|-- json_header.xlsx (optional)
```
Comment (Nora-Olivia-Ammann) [inline docs/file-formats/excel2json.md:36]: @jnussbaum something like this?

File: src/dsp_tools/commands/excel2json/project.py [hunk 5]
```diff
@@ -59,20 +60,22 @@ def excel2json(
...
             f"The only allowed subfolders in '{data_model_files}' are 'lists' "
             "and folders that match the pattern 'onto_name (onto_label)'"
         )
+    if (json_header := Path(data_model_files) / Path("json_header.xlsx")).exists():
+        files_to_process.append(str(json_header))
     print("The following files will be processed:")
-    print(*(f" - {file}" for file in processed_files), sep="\n")
-    return listfolder, onto_folders
+    print(*(f" - {file}" for file in files_to_process), sep="\n")
+    return listfolder_content, onto_folder_content
```
Comment (Nora-Olivia-Ammann) [inline src/dsp_tools/commands/excel2json/project.py:78]: Is it clearer now?

File: src/dsp_tools/commands/excel2json/project.py [hunk 6]
```diff
@@ -22,12 +22,13 @@ def excel2json(
...
-        |-- lists
-        |   |-- de.xlsx
-        |   `-- en.xlsx
-        `-- onto_name (onto_label)
-            |-- properties.xlsx
-            `-- resources.xlsx
+        ├── json_header.xlsx (optional)
+        ├── lists
+        │   ├── list.xlsx
+        │   └── list_1.xlsx
```
Comment (jnussbaum) [inline src/dsp_tools/commands/excel2json/project.py:28]: I think this is the old excel2json, so it should be like this:

```suggestion
        │   ├── de.xlsx
        │   └── en.xlsx
```

File: docs/file-formats/excel2json.md [hunk 7]
```diff
@@ -66,14 +66,13 @@ All the files containing "list" in their name will be read and processed, other
...
-|   `-- other_list.xlsx
-|   `-- list3.xlsx
-`-- onto_name (onto_label)
-    |-- properties.xlsx
-    `-- resources.xlsx
-    `json_header.xlsx` (optional)
+├── json_header.xlsx (optional)
+├── lists
+│   ├── list.xlsx
+│   └── list_1.xlsx
```
Comment (jnussbaum) [inline docs/file-formats/excel2json.md:72]: Not sure how important this is, but content-wise, you original was:

```suggestion
│   ├── lists.xlsx
│   ├── other_list.xlsx
│   └── other_list.xlsx
```

File: docs/file-formats/excel2json.md [hunk 8]
```diff
@@ -27,13 +27,13 @@ The following folder structure has to be used, when invoking the standard `excel
...
-|   |-- de.xlsx
-|   `-- en.xlsx
-`-- onto_name (onto_label)
-    |-- properties.xlsx
-    `-- resources.xlsx
-    `json_header.xlsx` (optional)
+├── json_header.xlsx (optional)
+├── lists
+│   ├── list.xlsx
+│   └── list_1.xlsx
```
Comment (jnussbaum) [inline docs/file-formats/excel2json.md:33]: For the old excel2json, it should be this:

```suggestion
│   ├── de.xlsx
│   └── en.xlsx
```

File: src/dsp_tools/commands/excel2json/project.py [hunk 9]
```diff
@@ -59,20 +60,22 @@ def excel2json(
...
-    folder = [x for x in Path(data_model_files).glob("*") if _non_hidden(x)]
-    processed_files = []
-    onto_folders, processed_onto = _get_and_validate_onto_folder(Path(data_model_files), folder)
-    processed_files.extend(processed_onto)
-    listfolder, processed_lists = _get_validate_list_folder(data_model_files, folder)
-    processed_files.extend(processed_lists)
-    if len(onto_folders) + len(listfolder) != len(folder):
+    sub_folders = [x for x in Path(data_model_files).glob("*") if _non_hidden(x) and x.is_dir()]
+    files_to_process = []
+    onto_folder_content, onto_to_process = _get_and_validate_onto_folder(Path(data_model_files), sub_folders)
```
Comment (jnussbaum) [inline src/dsp_tools/commands/excel2json/project.py:65]: question regarding onto_to_process: Are these the onto files to be processed? Then I suggest `onto_files_to_process`

File: src/dsp_tools/commands/excel2json/project.py [hunk 10]
```diff
@@ -59,20 +60,22 @@ def excel2json(
...
-    onto_folders, processed_onto = _get_and_validate_onto_folder(Path(data_model_files), folder)
-    processed_files.extend(processed_onto)
-    listfolder, processed_lists = _get_validate_list_folder(data_model_files, folder)
-    processed_files.extend(processed_lists)
-    if len(onto_folders) + len(listfolder) != len(folder):
+    sub_folders = [x for x in Path(data_model_files).glob("*") if _non_hidden(x) and x.is_dir()]
+    files_to_process = []
+    onto_folder_content, onto_files_to_process = _get_and_validate_onto_folder(Path(data_model_files), sub_folders)
+    files_to_process.extend(onto_files_to_process)
+    listfolder_content, lists_to_process = _get_validate_list_folder(data_model_files, sub_folders)
```
Comment (jnussbaum) [inline src/dsp_tools/commands/excel2json/project.py]: listfolder_content: This one is awkward: It is a list containing only 1 element, namely the list folder. This is misleading per se and cannot be improved just by renaming. So let's keep it for today, in order not to make this PR too big 😃

File: src/dsp_tools/commands/excel2json/project.py [hunk 11]
```diff
@@ -59,20 +60,22 @@ def excel2json(
...
-    folder = [x for x in Path(data_model_files).glob("*") if _non_hidden(x)]
-    processed_files = []
-    onto_folders, processed_onto = _get_and_validate_onto_folder(Path(data_model_files), folder)
-    processed_files.extend(processed_onto)
-    listfolder, processed_lists = _get_validate_list_folder(data_model_files, folder)
-    processed_files.extend(processed_lists)
-    if len(onto_folders) + len(listfolder) != len(folder):
+    sub_folders = [x for x in Path(data_model_files).glob("*") if _non_hidden(x) and x.is_dir()]
+    files_to_process = []
+    onto_folder_content, onto_files_to_process = _get_and_validate_onto_folder(Path(data_model_files), sub_folders)
```
[the remark this answers is not in the document]
Comment (Nora-Olivia-Ammann) [reply src/dsp_tools/commands/excel2json/project.py]: Better?

**Review by jnussbaum — APPROVED** [review]
LGTM

File: src/dsp_tools/commands/excel2json/project.py (hunk 10 above)
Comment (Nora-Olivia-Ammann) [reply src/dsp_tools/commands/excel2json/project.py]: I just changed it back

Comment (jnussbaum) [reply src/dsp_tools/commands/excel2json/project.py]: Okay, you already improved it. Yes, now it's better understandable for me than before. Thank you :-)

**Pull request merged**